### ISSUE TITLE
I test the code on mnist and another structured data

### ISSUE THREAD

**Issue by fresh2tensorflow** [pylablanche/gcForest#5]
but why my result show that the gcforest is not so powerfull when compared with cnn and xgboost，am I wrong somewhere？

**Comment by pylablanche** [comment]
Probably not.
Some people have already pointed out that gcForest does not always perform better or significantly better than cnn and xgboost (see discussino here : https://github.com/Microsoft/LightGBM/issues/331#issuecomment-307401782 )

How much accuracy do you get on the MNIST dataset?

**Comment by fresh2tensorflow** [comment]
thank you for you reply.
(1)the result of my experiment is in the pdf file attached.
(2)I don't understand the parameter tuning of gcforest，I just tune it as my thougth.
(3)when running your code, one place in GCForest.py report error,I change "getattr" to "hasattr", then it can run.(line 175 in GCForest.py)

(4)did you try the official code on the github several days ago?I run it I tried several time and report error.

thank you very much for your kindness.  :)







At 2017-06-15 13:47:03, "Pierre-Yves Lablanche" <notifications@github.com> wrote:


Probably not.
Some people have already pointed out that gcForest does not always perform better or significantly better than cnn and xgboost (see discussino here : Microsoft/LightGBM#331 (comment) )

How much accuracy do you get on the MNIST dataset?

**Comment by fresh2tensorflow** [comment]
thank you for you reply.
(1)the result of my experiment is in the pdf file attached.
(2)I don't understand the parameter tuning of gcforest，I just tune it as my thougth.
(3)when running your code, one place in GCForest.py report error,I change "getattr" to "hasattr", then it can run.(line 175 in GCForest.py)

(4)did you try the official code on the github several days ago?I run it I tried several time and report error.

thank you very much for your kindness.  :)







At 2017-06-15 13:47:03, "Pierre-Yves Lablanche" <notifications@github.com> wrote:


Probably not.
Some people have already pointed out that gcForest does not always perform better or significantly better than cnn and xgboost (see discussino here : Microsoft/LightGBM#331 (comment) )

How much accuracy do you get on the MNIST dataset?

**Comment by fresh2tensorflow** [comment]
thank you for you reply.
(1)the result of my experiment is:
      
|

Mnist

|

GCForest

|

CNN

|
|

machine

|

Tf5/cpu

|

PC/cpu

|
|

accuracy

|

79.28%

|

81.49%

|

82.98%

|

99.16%

|
|

time cost

|

21min

|

38min

|

3h15min

|

1h7min

|

(2)I don't understand the parameter tuning of gcforest，I just tune it as my thougth.
(3)when running your code, one place in GCForest.py report error,I change "getattr" to "hasattr", then it can run.(line 175 in GCForest.py)

(4)did you try the official code on the github several days ago?I run it I tried several time and report error.

thank you very much for your kindness.  :)







At 2017-06-15 13:47:03, "Pierre-Yves Lablanche" <notifications@github.com> wrote:


Probably not.
Some people have already pointed out that gcForest does not always perform better or significantly better than cnn and xgboost (see discussino here : Microsoft/LightGBM#331 (comment) )

How much accuracy do you get on the MNIST dataset?

**Comment by fresh2tensorflow** [comment]
*_*,so...I reply the email, and  it reply on the github,it looks intersing...

**Comment by fresh2tensorflow** [comment]
best accuracy of gcforest on mnist is 83%..which cost 3 hours to train.  for cnn,99% in 1 hour.

**Comment by pylablanche** [comment]
@fresh2tensorflow I am not so surprised about the accuracy if you used gcForest out of the box.
Have you tried to replicate the architecture from the paper ? (i.e. 1000 trees per forest, 2 forests of each type, etc.) Actually if you could share the parameters you used that could be valuable information for future users.

Did you also build your own ConvNet or was it an existing code ?

**Comment by fresh2tensorflow** [comment]
thank you for your reply :)
(1)the first question: I didn't do it..I will try it then
(2)the second question:I use the convnet of the demo in the official tensorflow tutorial.It is a very simple convnet with just 2 conv-layer and 1 dense layer..I think 99.2% is not a bad result,so I didn't  
tune the parameter to get a better result..



At 2017-06-17 18:22:54, "Pierre-Yves Lablanche" <notifications@github.com> wrote:


@fresh2tensorflow I am not so surprised about the accuracy if you used gcForest out of the box.
Have you tried to replicate the architecture from the paper ? (i.e. 1000 trees per forest, 2 forests of each type, etc.) Actually if you could share the parameters you used that could be valuable information for future users.

Did you also build your own ConvNet or was it an existing code ?

**Comment by Laurae2** [comment]
@fresh2tensorflow Try to use a subsample only so you can train faster and check performance difference faster.

Example on MNIST 1k samples:

![MNIST](https://cloud.githubusercontent.com/assets/9083669/24105619/be2e4a2a-0d85-11e7-8a1f-cf527308be48.png)

**Comment by fresh2tensorflow** [comment]
@Laurae2     I get it，thank you. :)

**Comment by pylablanche** [comment]
Thanks @Laurae2 ! Btw have you seen this : https://arxiv.org/pdf/1705.07366.pdf

@fresh2tensorflow Would you mind sharing your results on the (mini) MNIST dataset?
I'm curious to know what you get!!!

**Comment by fresh2tensorflow** [comment]
@pylablanche     oh,I am sorry, I didn't  try the algorithm on a mini MNIST..   I think it is a good idea when exploring the data..
and then I try the the gcforest with parameter(1000 trees per forest, 2 forests of each type),its accuracy didn't improve much..
